main.cpp

```diff
@@ -17,15 +17,16 @@ int main(int argc, char **argv)
         const char *filtersOptions {"begrs"};
 
         // Get filter flag
-        char filter {static_cast<char>(getopt(argc, argv, filtersOptions))};
+        unsigned char filter {static_cast<unsigned char>(getopt(argc, argv, filtersOptions))};
 
         // check validity
         if (filter == '?') {
             throw std::string {"Invalid filter."};
         }
 
         // Ensure only one filter
-        if (filter != -1) {
+        if (getopt(argc, argv, filtersOptions) != -1) {
+            std::cout << filter << std::endl;
```

Comment:
Remove `std::cout`. I only added in terms of testing. It should not be present in `main.cpp`.

Reply:
My bad..... After testing I forgot to remove it.